lib/bootloader_setup.pm

```diff
@@ -1348,16 +1348,20 @@ sub prepare_disks {
     my $disks = script_output('lsblk -n -l -o NAME -d -e 7,11');
     for my $d (split('\n', $disks)) {
         script_run "wipefs -af /dev/$d";
-        if (get_var('ENCRYPT_ACTIVATE_EXISTING') || get_var('ENCRYPT_CANCEL_EXISTING'))
-        {
+        script_run "sync";
+        if (get_var('ENCRYPT_ACTIVATE_EXISTING') || get_var('ENCRYPT_CANCEL_EXISTING')) {
             create_encrypted_part(disk => $d);
             if (get_var('ETC_PASSWD') && get_var('ETC_SHADOW')) {
                 mimic_user_to_import(disk => $d,
                     passwd => get_var('ETC_PASSWD'),
                     shadow => get_var('ETC_SHADOW'));
             }
+        } else {
+            script_run "parted /dev/$d mklabel gpt";
```

Comment:
I wonder if we should rather use initial approach and run dd command to write zeros to the disk. As with this command we setup the env like there was something on the disk and not completely empty disk.
In case you are fine to use `dd` command, I would also suggest to limit it to the cases when it's needed (e.g. ipmi), as it's rather slow due to number of I/Os to execute.

Also, this all looks like a bug in the product to me, is it reported?

Cheers!

Reply:
Hi Rodion, OK.
I have a day off tomorrow, but next week I will try something like `dd bs=1M count=20`.
Until then the issue will be marked as `NotReady`.

Reply:
Why not both? First nuke it with `dd` and then run `parted ... mklabel gpt`?

I'd argue that `mklabel` is a good approach, because it should also take care of the secondary GTP header at the end of the disk, which otherwise remains intact (See [Layout](https://en.wikipedia.org/wiki/GUID_Partition_Table)).